src/test/java/org/beanpod/switchboard/dao/StreamDaoImplTest.java

```diff
@@ -88,6 +91,121 @@ public void testCreateChannel() {
+
+    // when & then
+    assertThrows(
+        ExceptionType.StreamAlreadyExistsException.class,
+        () -> streamService.createStream(createStreamRequest));
+  }
+
+  @Test
+  public void testCreateChannelDifferentPublicIp() {
+    String otherPublicIpAddress = "179.256.54.21";
+    CreateStreamRequest createStreamRequest = StreamFixture.getCreateStreamRequest();
+    InputChannelDTO inputChannelDto = ChannelFixture.getInputChannelDto();
+    OutputChannelDTO outputChannelDto = ChannelFixture.getOutputChannelDto();
+    outputChannelDto.getEncoder().getDevice().setPublicIpAddress(otherPublicIpAddress);
+    StreamEntity streamEntity = StreamFixture.getStreamEntity();
+
+    when(channelService.getInputChannelById(createStreamRequest.getInputChannelId()))
+        .thenReturn(inputChannelDto);
+    when(channelService.getOutputChannelById(createStreamRequest.getOutputChannelId()))
+        .thenReturn(outputChannelDto);
+    when(streamMapper.toEntity(any())).thenReturn(streamEntity);
+    when(streamRepository.existsDuplicate(
+            createStreamRequest.getInputChannelId(), createStreamRequest.getOutputChannelId()))
+        .thenReturn(false);
+
+    // when
+    streamService.createStream(createStreamRequest);
+
+    // then
+    verify(streamRepository).save(streamEntity);
+  }
+
+  @Test
+  public void testCreateChannelSameDeviceIpV4() {
+    String otherPublicIpAddress = StreamDaoImpl.LOOPBACK_IP_V4;
+    CreateStreamRequest createStreamRequest = StreamFixture.getCreateStreamRequest();
+    InputChannelDTO inputChannelDto = ChannelFixture.getInputChannelDto();
+    OutputChannelDTO outputChannelDto = ChannelFixture.getOutputChannelDto();
+    outputChannelDto.getEncoder().getDevice().setPublicIpAddress(otherPublicIpAddress);
+    StreamEntity streamEntity = StreamFixture.getStreamEntity();
+
+    when(channelService.getInputChannelById(createStreamRequest.getInputChannelId()))
+        .thenReturn(inputChannelDto);
+    when(channelService.getOutputChannelById(createStreamRequest.getOutputChannelId()))
+        .thenReturn(outputChannelDto);
+    when(streamMapper.toEntity(any())).thenReturn(streamEntity);
+    when(streamRepository.existsDuplicate(
+            createStreamRequest.getInputChannelId(), createStreamRequest.getOutputChannelId()))
+        .thenReturn(false);
+
+    // when
+    streamService.createStream(createStreamRequest);
+
+    // then
+    verify(streamRepository).save(streamEntity);
+  }
+
+  @Test
+  public void testCreateChannelSameDeviceIpV6() {
+    String otherPublicIpAddress = StreamDaoImpl.LOOPBACK_IP_V6;
+    CreateStreamRequest createStreamRequest = StreamFixture.getCreateStreamRequest();
+    InputChannelDTO inputChannelDto = ChannelFixture.getInputChannelDto();
+    OutputChannelDTO outputChannelDto = ChannelFixture.getOutputChannelDto();
+    outputChannelDto.getEncoder().getDevice().setPublicIpAddress(otherPublicIpAddress);
+    StreamEntity streamEntity = StreamFixture.getStreamEntity();
```

Comment:
i think we can put these in before each?

Reply:
Yep, I moved them.